model/plugin_event.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+const (
+	PluginEventSendTypeReliable   = CLUSTER_SEND_RELIABLE
+	PluginEventSendTypeBestEffort = CLUSTER_SEND_BEST_EFFORT
+)
+
+// PluginEvent is used to allow intra-cluster plugin communication.
+type PluginEvent struct {
```

Comment:
I decided to go with a dedicated structure mainly to try and abstract a bit from the details of our cluster implementation.